livekit-plugins/livekit-plugins-neuphonic/livekit/plugins/neuphonic/tts.py

```diff
@@ -0,0 +1,434 @@
+
+        if not api_key:
+            raise ValueError("NEUPHONIC_API_TOKEN must be set")
+
+        self._opts = _TTSOptions(
+            model=model,
+            voice_id=voice_id,
+            lang_code=lang_code,
+            encoding=encoding,
+            speed=speed,
+            sampling_rate=sample_rate,
+            api_key=api_key,
+            base_url=base_url,
+        )
+
+        self._session = http_session
+        self._pool = utils.ConnectionPool[aiohttp.ClientWebSocketResponse](
+            connect_cb=self._connect_ws,
+            close_cb=self._close_ws,
+            max_session_duration=90,
+            mark_refreshed_on_get=True,
+        )
+        self._streams = weakref.WeakSet[SynthesizeStream]()
+
+    async def _connect_ws(self) -> aiohttp.ClientWebSocketResponse:
+        session = self._ensure_session()
+        url = f"wss://{self._opts.base_url}/speak/{self._opts.lang_code}{self._opts.get_query_param_string()}"
+
+        return await asyncio.wait_for(
+            session.ws_connect(url, headers={AUTHORIZATION_HEADER: self._opts.api_key}),
+            self._conn_options.timeout,
+        )
+
+    async def _close_ws(self, ws: aiohttp.ClientWebSocketResponse):
+        await ws.close()
+
+    def _ensure_session(self) -> aiohttp.ClientSession:
+        if not self._session:
+            self._session = utils.http_context.http_session()
+
+        return self._session
+
+    def prewarm(self) -> None:
+        self._pool.prewarm()
+
+    def update_options(
+        self,
+        *,
+        model: TTSModels | str = None,
+        voice_id: str | None = None,
+        lang_code: TTSLangCodes | str | None = None,
+        encoding: TTSEncodings | str | None = None,
+        speed: float | None = None,
+        sample_rate: int | None = None,
+    ) -> None:
+        """
+        Update the Text-to-Speech (TTS) configuration options.
+
+        This method allows updating the TTS settings, including model type, voice_id, lang_code, 
+        encoding, speed and sample_rate. If any parameter is not provided, the existing value will be 
+        retained.
+
+        Args:
+            model (TTSModels | str, optional): The Neuphonic model to use.
+            voice_id (str, optional): The voice ID for the desired voice.
+            lang_code (TTSLanguages | str, optional): The language code for synthesis..
+            encoding (TTSEncodings | str, optional): The audio encoding format.
+            speed (float, optional): The audio playback speed.
+            sample_rate (int, optional): The audio sample rate in Hz.
+        """
+        self._opts.model = model or self._opts.model
+        self._opts.voice_id = voice_id or self._opts.voice_id
+        self._opts.lang_code = lang_code or self._opts.lang_code
+        self._opts.encoding = encoding or self._opts.encoding
+        self._opts.speed = speed or self._opts.speed
+        self._opts.sampling_rate = sample_rate or self._opts.sampling_rate
```

Comment:
```suggestion
        self._opts.sampling_rate = sample_rate or self._opts.sampling_rate
        self._pool.invalidate()
```
it looks like neuphonic sets options upon connection, so we need to invalidate the pool to get a new connection with the updated options

Reply:
Good spot!